### PR TITLE
Port to conan

### PR DESCRIPTION
**Pull request by Shaheen47**
Referring to issue #68 , I created a conan recipe for the project .
i did some small change in the project structures(moved the headers into an include folder and created a main CMakeList.txt which is required for the conan recipe).
i noticed that you are using boost.optional in one of your header files but leeting the user include boost package for using it...we can also use conan to include the dependency of boost if you wish(i havent done it yet).
Now you have to create an account in  Artifactory in order to publish your package for public.
please check https://docs.conan.io/en/latest/uploading_packages.html to see how to create the package from the conan recipe i have created for this project .
please also check the conanfile.py file so you can change some meta information related to your project there

### CODE REVIEW

**Comment by Shaheen47** [comment]
please also check the conanfile.py file so you can change some meta information related to your project there

**Comment by ryanhaining** [comment]
moving to an `include/` is required by conan? or creating a main `CMakeList.txt` is required? Moving to an include is gonna break everyone syncing the repo (along with travis + appveyor)

**Comment by Shaheen47** [comment]
moving to an include/ is not required by conan ,it is just for including a folder instead of writing all the headers names in the CmakeList and in the conan recipe.so it is not a mandatory step and i can create the recipe without it .
However,to the best of my knowledge(i might be wrong) the CMakeList.txt is necessary .

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+using namespace std::string_literals;
+
+int main() {
+    bool value = true;
+    while(value)
+        value += true;
+    std::cout << value << std::endl;
+}
```
Comment (ryanhaining) [inline main.cpp]: what is this?

File: include/combinations_with_replacement.hpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #ifndef ITER_COMBINATIONS_WITH_REPLACEMENT_HPP_
 #define ITER_COMBINATIONS_WITH_REPLACEMENT_HPP_
 
-#include "internal/iteratoriterator.hpp"
+  #include "internal/iteratoriterator.hpp"
```
Comment (ryanhaining) [inline include/combinations_with_replacement.hpp]: remove this extra space please

**Comment by Shaheen47** [comment]
i will delete this pull request and create another one with all the changes you requested .

**Pull request closed without merging**